zipkin-thrift/src/main/thrift/com/twitter/zipkin/zipkinCore.thrift

```diff
@@ -15,18 +15,118 @@ namespace java com.twitter.zipkin.thriftjava
+ */
 const string CLIENT_RECV = "cr"
+/*
+ * The server sent ("ss") a response to a client. There is only one response
+ * per span. If there's a transport error, each attempt can be logged as a
+ * WIRE_SEND annotation.
+ *
+ * Typically, a trace ends with a server send, so the last timestamp of a trace
+ * is often the timestamp of the root span's server send.
+ *
+ * If chunking is involved, each chunk could be logged as a separate
+ * SERVER_SEND_FRAGMENT in the same span.
+ *
+ * Annotation.host is not the client. It is the host which logged the send
+ * event, almost always the server.
+ *
+ * To record the actual endpoints of the client and server side of the span,
+ * use CLIENT_ADDR and SERVER_ADDR respectively.
+ */
 const string SERVER_SEND = "ss"
+/*
+ * The server received ("cr") a request from a client. There is only one
+ * request per span.  For example, if duplicate responses were received, each
+ * can be logged as a WIRE_RECV annotation.
+ *
+ * Typically, a trace starts with a server receive, so the first timestamp of a
+ * trace is often the timestamp of the root span's server receive.
+ *
+ * If chunking is involved, each chunk could be logged as a separate
+ * SERVER_RECV_FRAGMENT in the same span.
+ *
+ * Annotation.host is not the client. It is the host which logged the send
+ * event, almost always the server.
+ *
+ * To record the actual endpoints of the client and server side of the span,
+ * use CLIENT_ADDR and SERVER_ADDR respectively.
+ */
 const string SERVER_RECV = "sr"
+/*
+ * Optionally logs an attempt to send a message on the wire. Multiple wire send
+ * events could indicate network retries. A lag between client or server send
+ * and wire send might indicate queing or processing delay.
+ */
 const string WIRE_SEND = "ws"
+/*
+ * Optionally logs an attempt to receive a message from the wire. Multiple wire
+ * receive events could indicate network retries. A lag between wire receive
+ * and client or server receive might indicate queing or processing delay.
+ */
 const string WIRE_RECV = "wr"
+/*
+ * Optionally logs progress of a (CLIENT_SEND, WIRE_SEND). For example, this
+ * could be one chunk in a chunked request.
+ */
+const string CLIENT_SEND_FRAGMENT = "csf"
+/*
+ * Optionally logs progress of a (CLIENT_RECV, WIRE_RECV). For example, this
+ * could be one chunk in a chunked response.
+ */
+const string CLIENT_RECV_FRAGMENT = "crf"
+/*
+ * Optionally logs progress of a (SERVER_SEND, WIRE_SEND). For example, this
+ * could be one chunk in a chunked response.
+ */
+const string SERVER_SEND_FRAGMENT = "ssf"
+/*
+ * Optionally logs progress of a (SERVER_RECV, WIRE_RECV). For example, this
+ * could be one chunk in a chunked request.
+ */
+const string SERVER_RECV_FRAGMENT = "srf"
 
-#************** Common binary annotation keys **************
-// The endpoint associated with CLIENT_ annotations is not necessarily Annotation.host
+#***** BinaryAnnotation.key where value = [1] and annotation_type = BOOL ******
```

Comment:
PS if in a new model we made endpoint a value type, we might get away with
not logging it as "host" all the time. I suspect the original decision of
CA/SA as a marker annotation was because the value are thrift primitive
types, not structs.

Also, by doing a marker type the normal
annotation.host/binaryannotation.host fields and related logic could be
reused.

Regardless, this is a great idea to bear in mind for model 2.0. I like the
idea of logging the protocol and also revisiting to not constrain to ipv4.